src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1072 @@
+            return mStorage->mSubjects[index].Get(subject);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR SetSubject(size_t index, NodeId subject) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        if (index < EntryStorage::kMaxSubjects)
+        {
+            return mStorage->mSubjects[index].Set(subject);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR AddSubject(size_t * index, NodeId subject) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        size_t count;
+        GetSubjectCount(count);
+        if (count < EntryStorage::kMaxSubjects)
+        {
+            CHIP_ERROR err = mStorage->mSubjects[count].Add(subject);
+            if (err == CHIP_NO_ERROR && index != nullptr)
+            {
+                *index = count;
+            }
+            return err;
+        }
+        return CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+
+    CHIP_ERROR RemoveSubject(size_t index) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        size_t count;
+        GetSubjectCount(count);
+        if (index < count)
+        {
+            const auto & dest = mStorage->mSubjects + index;
+            const auto & src  = dest + 1;
+            auto bytes        = (count - index - 1) * sizeof(*dest);
+            memmove(dest, src, bytes);
+            if (count == EntryStorage::kMaxSubjects)
+            {
+                mStorage->mSubjects[EntryStorage::kMaxSubjects - 1].Clear();
+            }
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR GetTargetCount(size_t & count) const override
+    {
+        count = 0;
+        for (const auto & target : mStorage->mTargets)
+        {
+            if (target.IsEmpty())
+            {
+                break;
+            }
+            count++;
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetTarget(size_t index, Target & target) const override
+    {
+        if (index < EntryStorage::kMaxTargets)
+        {
+            return mStorage->mTargets[index].Get(target);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR SetTarget(size_t index, const Target & target) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        if (index < EntryStorage::kMaxTargets)
+        {
+            return mStorage->mTargets[index].Set(target);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR AddTarget(size_t * index, const Target & target) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        size_t count;
+        GetTargetCount(count);
+        if (count < EntryStorage::kMaxTargets)
+        {
+            CHIP_ERROR err = mStorage->mTargets[count].Add(target);
+            if (err == CHIP_NO_ERROR && index != nullptr)
+            {
+                *index = count;
```

Comment:
Yeah, this is _not_ the semantics I was expecting for that index pointer arg.... Needs API documentation in the pure virtual class.

Reply:
Yes documentation is coming. My first priority is getting this module to unblock downstream development, my second priority is more test cases (for fabric indexing, which probably has a bug, but doesn't block downstream development), then documentation.